partiql-eval/src/main/java/org/partiql/eval/PQLValue.java

```diff
@@ -0,0 +1,615 @@
+                return new IntervalValue(Objects.requireNonNull(INTERVALValue.getValue()));
+            case TIMESTAMP:
+                org.partiql.value.TimestampValue TIMESTAMPValue = (org.partiql.value.TimestampValue) value;
+                return new TimestampValue(Objects.requireNonNull(TIMESTAMPValue.getValue()));
+            case TIME:
+                org.partiql.value.TimeValue TIMEValue = (org.partiql.value.TimeValue) value;
+                return new TimeValue(Objects.requireNonNull(TIMEValue.getValue()));
+            case FLOAT32:
+                org.partiql.value.Float32Value FLOAT32Value = (org.partiql.value.Float32Value) value;
+                return new Float32Value(Objects.requireNonNull(FLOAT32Value.getValue()));
+            case FLOAT64:
+                org.partiql.value.Float64Value FLOAT64Value = (org.partiql.value.Float64Value) value;
+                return new Float64Value(Objects.requireNonNull(FLOAT64Value.getValue()));
+            case DECIMAL:
+                org.partiql.value.DecimalValue DECIMALValue = (org.partiql.value.DecimalValue) value;
+                return new DecimalValue(Objects.requireNonNull(DECIMALValue.getValue()));
+            case CHAR:
+                org.partiql.value.CharValue CHARValue = (org.partiql.value.CharValue) value;
+                return new CharValue(Objects.requireNonNull(Objects.requireNonNull(CHARValue.getValue()).toString()));
+            case SYMBOL:
+                org.partiql.value.SymbolValue SYMBOLValue = (org.partiql.value.SymbolValue) value;
+                return new SymbolValue(Objects.requireNonNull(SYMBOLValue.getValue()));
+            case CLOB:
+                org.partiql.value.ClobValue CLOBValue = (org.partiql.value.ClobValue) value;
+                return new ClobValue(Objects.requireNonNull(CLOBValue.getValue()));
+            case BLOB:
+                org.partiql.value.BlobValue BLOBValue = (org.partiql.value.BlobValue) value;
+                return new BlobValue(Objects.requireNonNull(BLOBValue.getValue()));
+            case BYTE:
+                org.partiql.value.ByteValue BYTEValue = (org.partiql.value.ByteValue) value;
+                return new ByteValue(Objects.requireNonNull(BYTEValue.getValue()));
+            case DECIMAL_ARBITRARY:
+                org.partiql.value.DecimalValue DECIMAL_ARBITRARYValue = (org.partiql.value.DecimalValue) value;
+                return new DecimalArbitraryValue(Objects.requireNonNull(DECIMAL_ARBITRARYValue.getValue()));
+            case ANY:
+            default:
+                throw new NotImplementedError();
+        }
+    }
+
+    @NotNull
+    static PQLValue nullValue() {
+        return new NullValue();
+    }
+
+    @NotNull
+    static PQLValue missingValue() {
+        return new MissingValue();
+    }
+
+    @NotNull
+    static PQLValue nullValue(@NotNull PartiQLValueType type) {
+        return new NullValue(type);
+    }
```

Comment:
In order to pass existing tests, this is necessary. I can mark it as deprecated, but for this PR I'd like to avoid removing PartiQLValueType.NULL as it touches many things

Reply:
See PR #1463. This will have a pretty significant impact on this comment.